File: Assignment12/Program1.c


//Problem Statement :Write a program which accept number from user and return the count of Even digits
#include<stdio.h>


//Function to return Even count in Number 
int CountEven(int iNo)
{
    auto  int iDigit = 0;
    auto int iCount = 0;
    
    
      if(iNo < 0)  //Updator
    {
        iNo = -iNo;
    }
     //joparient iNo zero hot nai toparient loop phirnar 
    while(iNo > 0)   //Condition 
    {
        iDigit = iNo % 10;  //   
        
        if(iDigit % 2 == 0) //Logic to find Even Digit 

        {
          iCount++;  //To count Even digit ,initially count initialise by zero(int count = 0)

        }
        iNo = iNo / 10; 
    }
    return iCount;  //Function reutrn icount to caller function 
}

int main()
{
    auto int iValue = 0; //Variable to store Number 
    auto int iRet = 0; //Varial to store Result

    printf("Enter number :"); //To display on screen
    scanf("%d",&iValue);      

    iRet = CountEven(iValue);//Function call

    printf("%d" , iRet); //To print Even Count of Number 

    return 0; //Return zero to os 

}

//Test Steps
/*
  Input : 2395
  Output :1

  Input :1018
  Output: 2

  INput : -1018
  Output :2
 
  iNput : 8462
  Output:4

*/

File: Assignment34/Program1.java


//1.Write a java Program which accept number from user and return the count of Even Digits
/*
//Input 2395
//Output 1
 
Input 1018
Output 2

Input -1018
Output 2

Input 8462
Output 4

*/
import java.util.*;


class Program1
{
    public static int CountEven(int iNo)
    {
        int iDigit = 0;
        int iCount = 0;
          
        while(iNo != 0)
        {
            iDigit = iNo % 10;
            if((iDigit % 2) == 0)
            {
                iCount++;
            }
            iNo = iNo / 10;
            
           
        }
        return iCount;
    }
    

    public static void main(String Args[])
    {
        Scanner sobj = new Scanner(System.in);
        int iValue = 0, iRet = 0;

        System.out.println("Enter Number :");
        iValue = sobj.nextInt();
        
        iRet = CountEven(iValue);
        System.out.println("Even Count is : "+iRet);
    }
}
